src/nsls2api/cli/api.py

```diff
@@ -1,13 +1,185 @@
+import httpx
 import typer
+from rich.box import ROUNDED
+from rich.console import Console
+from rich.panel import Panel
+from rich.table import Table
+from rich.text import Text
+from rich.theme import Theme
 
-app = typer.Typer()
+from nsls2api.cli.settings import get_base_url, get_token
+from nsls2api.cli.utils.cli_helpers import auto_help_if_no_command
+
+app = typer.Typer(invoke_without_command=True)
+
+
+@app.callback()
+@auto_help_if_no_command()
+def api_callback(ctx: typer.Context):
+    pass  # No need to call anything manually
+
+
+console = Console(
+    theme=Theme(
+        {
+            "info": "cyan",
+            "warning": "yellow",
+            "error": "red bold",
+            "success": "green bold",
+            "health.good": "green bold",
+            "health.bad": "red bold",
+        }
+    )
+)
 
 
 @app.command()
 def status():
-    print("General status of NSLS-II API")
+    """Show general status of NSLS-II API"""
+    url = f"{get_base_url()}/v1/about"
+
+    with console.status("[cyan]Fetching API status...", spinner="dots"):
+        try:
+            with httpx.Client() as client:
+                token = get_token()
+                headers = {"Authorization": f"{token}"} if token else {}
+                response = client.get(url, headers=headers)
+                response.raise_for_status()
+                about_data = response.json()
+
+                # Create status table
+                status_table = Table(show_header=False, box=None)
+                status_table.add_column("Key", style="cyan")
+                status_table.add_column("Value", style="green")
+
+                status_table.add_row("Status", "✓ Available")
+                status_table.add_row("Version", about_data["version"])
+                status_table.add_row("Description", about_data["description"])
+                status_table.add_row("API URL", get_base_url())
+
+                # Add container info if available
+                if "container_info" in about_data and about_data["container_info"]:
```

Comment:
This might just be a style preference, but you could use the simpler idiomatic approach here:
```suggestion
                if about_data.get("container_info"):
```